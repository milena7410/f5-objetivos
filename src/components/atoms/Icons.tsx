import * as React from "react";
import Feather from "@expo/vector-icons/Feather";
import Octicons from "@expo/vector-icons/Octicons";
import MaterialIcons from "@expo/vector-icons/MaterialIcons";

import { cssInterop } from "nativewind";

const IconTypes = {
  materialIcons: MaterialIcons,
  feather: Feather,
  octicons: Octicons,
};

type CheckListIconProps = {
  type: keyof typeof IconTypes;
  name:
    | "check-box"
    | "check-square"
    | "checklist"
    | "edit"
    | "home"
    | "square"
    | "trash-2";
  size?: number;
  className?: string;
  style?: {};
  color?: string;
};

function extractCss(component: React.FC<CheckListIconProps>) {
  return cssInterop(component, {
    className: {
      target: "style",
      nativeStyleToProp: {
        color: true,
        height: "size",
        width: "size",
      },
    },
  });
}

export const Icons = extractCss(
  ({ name, type, className, ...rest }: CheckListIconProps) => {
    const IconLib = IconTypes[type];
    return <IconLib name={name as any} {...rest} />;
  }
);
