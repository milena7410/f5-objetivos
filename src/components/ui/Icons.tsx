import Feather from "@expo/vector-icons/Feather";
import { cssInterop } from "nativewind";

type CheckListIconProps = {
  size?: number;
  className?: string;
  style?: {};
  color?: string;
};

export const CheckListIcon = cssInterop(
  ({ ...rest }: CheckListIconProps) => {
    console.log(rest);
    return <Feather name="check-square" {...rest} />;
  },
  {
    className: {
      target: "style",
      nativeStyleToProp: {
        color: true,
        height: "size",
        width: "size",
      },
    },
  }
);
