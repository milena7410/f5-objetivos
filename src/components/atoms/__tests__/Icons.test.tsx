import React from "react";
import { render } from "@testing-library/react-native";

import { Icons } from "../Icons";

describe("Icons Component", () => {
  it("should render Icon when Feather", async () => {
    const { toJSON } = render(
      <Icons
        className="text-primary size-5"
        type="feather"
        name="check-square"
      />
    );
    expect(toJSON).toMatchSnapshot();
  });

  it("should render Icon when Octicons", async () => {
    const { toJSON } = render(
      <Icons className="text-primary size-5" type="octicons" name="home" />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
