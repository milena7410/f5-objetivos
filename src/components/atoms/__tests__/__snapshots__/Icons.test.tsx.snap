// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons Component should render Icon when Feather 1`] = `[Function]`;

exports[`Icons Component should render Icon when Octicons 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    [
      {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      {
        "fontFamily": "octicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      {},
    ]
  }
>
  ï†„
</Text>
`;
